mf4_csv/main.py

```diff
@@ -0,0 +1,52 @@
+
+import os 
+from asammdf import MDF, Signal
+from os import walk
+import re
+import glob
+
+
+output_dir = "./output_in_csv"
+
+def prepare_dir(mf4_file):
+    my_str = str(mf4_file)
+    my_str = my_str.lower().strip(".mf4")
+    my_str = my_str.replace(".MF4", ".csv")
+    # chars = "./-\"
+    # for c in chars:
+    #     text = text.replace(c, "\\" + c)
+    rx = re.compile('([-./\ ])')
+    my_str = rx.sub('_', my_str)
+    # print(my_str)
+    return my_str
+
+   
+
+def convertMF4toCSV(mf4_file):
+    data = MDF(mf4_file)
+    # f_out = f_in.replace(".MF4", ".csv") 
+    # mdf = MDF(f_in) #read mf4 data 
+
+    # d_dir = prepare_dir(mf4_file)
+    # print(d_dir)
+    f_out = mf4_file.replace(".MF4", ".csv")
+    data.export(fmt='csv', filename=f_out) #write to csv 
+    print("{} group completed".format(f_out))
+
+
+for root, dirs, files in os.walk("./"):
```

Comment:
Can this be moved into a python main function with a main guard? In the future this script could be included in something more high level that does batch work and being able to ensure the correct operation of the file will be nice.

This is some general information on python main guard https://stackoverflow.com/a/419185

Reply:
I will look into that tomorrow. I have HW to submit now.